### ISSUE TITLE
"Analysis of all end-to-end flows in a system instance" verification fails

### ISSUE THREAD

**Issue by paolo-crisafulli** [osate/alisa#56]
Hi,

I have created a simple model with associated Alisa files, that you can find here:
https://gist.github.com/paolo-crisafulli/8310d73e3af3d35f24e8fedf4b7b58a2

This model contains two chained processes with flows and latencies, as well as and an end-to-end flow traversing the two processes, with a latency spec, inferior to the sum of the two processes latencies.

Therefore, when a latency analysis is performed, as expected I have end-to-end latencies errors reported.

But when I run the Alisa verification, the corresponding evidence "responsetime" passes.

The documentation of FlowLatencyAnalysis method (in project AlisaPredefined) states "Analysis of all end-to-end flows in a system instance or for a specific end-to-end flow".

Shouldn't Alisa report a claim violation here?

**Comment by reteprelief** [comment]
Hi Paolo,

You latency requirement is written for the component and does not identify the end to end flow it applies to. Therefore, when retrieving the results from the latency analysis ALISA does not know which end to end flow results you are interested in.
You need to add a “for end_to_end” to your requirement.
Peter
      requirement R1 : "response time limit" for end_to_end [
             description this " shall have a global computation time within specified bounds"
      ]

**Comment by paolo-crisafulli** [comment]
Hi Peter,
Thank you for your answer.
I was aware of this approach, which I related to the "for a specific end-to-end flow" of the documentation of the FlowLatencyAnalysis method.
I was expecting to get an overall verification of all end-to-end flows of the component if no particular e2e flow was specified (the "all end-to-end flows in a system instance" part of the method documentation).
Could you explain the semantics of the method if no flow is specified? Does it mean that it verifies nothing in that case?
Thank you very much in advance.

**Comment by paolo-crisafulli** [comment]
Put another way: 
- wouldn't it make sense that if no particular e2e flow is mentioned, all of them are assessed ;
- or a syntax error is raised ?

**Comment by reteprelief** [comment]
You are right: we should either let the user know (I added a diagnostic message.
We are working on a way to handle "all" - in particular in support of global/reusable requirements in ALISA.
In that context an interesting question is what does "all" mean for a given component instance:
* all end to end flows directly contained in that component, or all end to end flows in that component and its subcomponents.
When you run the latency analysis manually it analyzes all end to end flows in the instance model.

**Comment by paolo-crisafulli** [comment]
Hi Peter,
Thank you for your answer and sorry mine comes so late, I've been away from my computer for quite a while.
Maybe all flows in a component and its subcomponents would be ok, as it can be controlled through the scope of the assurance plan ("assure subsystem all" or not)?

**Comment by paolo-crisafulli** [comment]
Hi @reteprelief,
Should I report this issue to osate/osate2 project for follow-up?
